### DISCUSSION TITLE
QFieldCloud Login failed

### DISCUSSION THREAD

**Discussion by SimonHHaas** [opengisch/QField#2526] [Q&A]
Hi,

I'm encountering a problem when trying to log into my QFieldCloud account in QGIS (3.22.3 on Windows 10 Pro). I don't know whether it is a software bug or a mistake on my side. I have tried restarting my computer, re-installing QGIS, re-installing the QField Sync Plugin, as well as changing my password to different lengths. 

So far I haven't been able to find a solution/explanation to this problem, so I would be glad if somebody could help me fix it.

<img width="330" alt="LoginFailed" src="https://user-images.githubusercontent.com/99903448/154537851-f670f996-9b30-432f-9741-c53e669ac490.png">

**Comment by suricactus** [comment]
Hey, @SimonHHaas .

I think I have managed to track the cause for your error. Can you drop us an email at `hello [at] qfield.cloud` so we can send you a QFieldSync version that fixes the issue?

Best,
Ivan

**Reply by gi921120** [reply]
Hi Ivan, 

I have a similar issue. I've just sent an email to hello@qfield.cloud. Could you please confirm me if this is the right email. 

Regards, 

Giovanny

**Reply by suricactus** [reply]
@gi921120 @SimonHHaas if you can test this QFieldSync version would be great.

[qfieldsync.master.zip](https://github.com/opengisch/QField/files/8095780/qfieldsync.master_fix_url.zip)

**Reply by SimonHHaas** [reply]
Hi @suricactus,

Unfortunately, this plugin also doesn't work for me. Is there perhaps something that I could've done wrong while setting it up/trying to log in?

Best,
Simon

**Reply by laotzu69** [reply]
Hi

I am also having the same issue - I have only recently been invited to try QFieldCloud but have not been able to login yet.

Best regards,
Antony

**Reply by suricactus** [reply]
To summarize, all of you:

1) have been invited recently to QFieldCloud.
2) successfully registered and managed to login using the web ui at app.qfield.cloud;
3) you are all using QGIS on Windows;
4) you have tried to login using QFieldSync BETA18 and the zipped QFieldSync above. It starts, but when you enter your correct credentials, you get a "login failed" feedback.

If this is the case, may I ask you for two favors:

1) enable the "debugging/development tools" panel from QGIS: View->Panels and grab a screenshot after your try to login. You should see something like this:
![image](https://user-images.githubusercontent.com/2820439/154687703-f4db298a-d09c-4eb3-a95d-d0d563cf6968.png)
2) try with an older version of QFieldSync, e.g. beta17 or beta16 from here https://plugins.qgis.org/plugins/qfieldsync/ .

Thank you for testing QFieldSync beta!

**Reply by SimonHHaas** [reply]
Hi @suricactus,

Your summary is correct in my case. 

I've tried using older versions of QFieldSync, but the results stay the same. 

Note: 'api/v1/' appears twice in the login URL, which seems a bit suspicious. When entering the URL from my screenshot into a browser, it says: Page is not found (404). However, the URL without the second 'api/v1/' seems to exist.

<img width="1022" alt="QFieldSyncLog" src="https://user-images.githubusercontent.com/99903448/154695974-499155e8-0488-496c-b244-e6e8e60628e5.png">

**Reply by suricactus** [reply]
Hi, @SimonHHaas . Thank you a lot! This is exactly what I suspected and tried to prevent in the qfieldsync version I attached above. I will post another solution very soon, thanks for your cooperation!

**Comment by laotzu69** [comment]
Hi 

My case is slightly different in that I successfully registered and have logged on via my Android device, but cannot log on again via the the web ui at app.qfield.cloud.
Thanks

    On Friday, 18 February 2022, 13:02:40 GMT, Ivan Ivanov ***@***.***> wrote:  
 
 


To summarize, all of you:
   
   - have been invited recently to QFieldCloud.
   - successfully registered and managed to login using the web ui at app.qfield.cloud;
   - you are all using QGIS on Windows;
   - you have tried to login using QFieldSync BETA18 and the zipped QFieldSync above. It starts, but when you enter your correct credential, you get a "login failed" feedback.

If this is the case, may I ask you for two favors:
   
   - enable the "debugging/development tools" panel from QGIS: View->Panels and grab a screenshot after your try to login. You should see something like this:   

   - try with an older version of QFieldSync, e.g. beta17 or beta16 from here https://plugins.qgis.org/plugins/qfieldsync/ .

Thank you for testing QFieldSync beta!

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Jimzi** [comment]
Hi,

Having the same issue.
This is the result from the Debugger panel.
I'm using 4.0.0-BETA18. Will try to downgrade to older versions as suggested above and post the results.

_Update:_ Tried downgrading to BETA17, BETA16, BETA15 (login failed on all three), and finally BETA14 (which seems to be where the cloud functionality was still not introduced).
Looking at the API path:
`Error: Error transferring https://app.qfield.cloud/api/v1/api/v1/auth/login/ - server replied: Not Found`
It seems that `/api/v1` is repeated twice.

Best, 
Jimmy

![Screenshot 2022-02-21 170120](https://user-images.githubusercontent.com/16478283/154981238-20b4dba2-ab3c-4057-a71a-0938d3c7e515.png)

**Comment by suricactus** [comment]
@SimonHHaas @Jimzi @laotzu69 @gi921120 @joekbullard (from https://github.com/opengisch/qfieldsync/issues/388)

Please check the latest QFieldSync BETA 19, it should fix your issues. Sorry for the inconvenience. Thanks for testing QFieldCloud and QFieldSync during beta phase, your feedback is invaluable.

**Reply by SimonHHaas** [reply]
Hi @suricactus,

Updating the QFieldSync plugin to beta19 did fix the problem. Thanks a lot for your help!

**Comment by m-lmi** [comment]
I suspect a hidden bug in QField Cloud when the user syncs data both from QField Sync plugin and QField app. 
I had it happen to multiple users that after they sync a project from QField Sync plugin in QGIS, and try to login in the QField app with the same username, the login fails (username and password are correct). 

The problem can be solved by clearing all data from the QField app. Clearing app cache is not enough. 

Can it be that the previously described bug in this thread is connected to this problem, but by reinstalling the plugin or app the app data gets cleared and therefore it solves the problem, without actually solving the underlying problem?

Can someone reproduce this?

Disclaimer: I run our on-prem QField Cloud

**Reply by suricactus** [reply]
@m-lmi not sure if I can reproduce. Can you please open a proper issue describing steps to reproduce. Also, do you run QField 4.2.4+ , if not, please check https://github.com/opengisch/QField/commit/3691fef841b11314b3a384b1e7743d83f36e5d4f .
